pkg/txn/client/operator.go

```diff
@@ -366,7 +367,18 @@ func (tc *txnOperator) updateWritePartitions(requests []txn.TxnRequest, locked b
 		defer tc.mu.Unlock()
 	}
 
+	// When session creates temporary tables, there may be cross-engine commit transactions
```

Comment:
Why you need change the txn operator?

Reply:
Now it is reusing the existing txnClient and txnOperator to operate the temporary engine. 

But commit may still be across engines.  We need to make sure that the temporary engine commit is done only after the existing engine commit is successful. So here we manually ensure that the TxnRequest for the temporary table is processed last.

The order of the existing TxnRequest has not been changed.